core/api/src/main/java/org/trellisldp/api/MutableDataService.java

```diff
@@ -16,50 +16,41 @@
 
 import java.util.concurrent.CompletableFuture;
 
-import org.apache.commons.rdf.api.Dataset;
 import org.apache.commons.rdf.api.IRI;
 
 /**
  * A service that persists resources by <i>replacing</i> their records.
  *
  * @author ajs6f
  * @param <U> the type of resource that can be persisted by this service
+ * @param <V> the type of data used as a template for generating a resource
  */
-public interface MutableDataService<U> extends RetrievalService<U> {
+public interface MutableDataService<U, V> extends RetrievalService<U> {
```

Comment:
I'm a bit uncomfortable that we have these two params that are clearly attached to each other semantically, but not in the type system at all. (Nothing prevents me from declaring a `MutableDataService<Resource, BinaryTemplate>`.)

Maybe we could do `Resource < ResourceTemplate` and the same for binaries? Or we could even think about a builder pattern...